### PR TITLE
fix(github): strip UTF-8-encoded C1 control sequences in Sanitize

### PR DESCRIPTION
**Pull request by gfazioli**
## What

Hardens `Sanitize` (octoscope's terminal-injection boundary) to also strip
UTF-8-encoded **C1 control characters** (U+0080–U+009F) — the 8-bit
CSI/OSC/DCS introducers that `ansi.Strip` (which only handles the 7-bit
ESC-prefixed forms) leaves intact.

This closes a doc-vs-code gap surfaced while adding the `Sanitize` tests
(#23): the docstring claimed C1 controls in multi-byte UTF-8 were "already
covered by the ansi pass" — they were not. A UTF-8-encoded U+009B
(bytes `0xC2 0x9B`) passed straight through both `ansi.Strip` and the byte
filter (both bytes are ≥ 0x80).

## Change

- `internal/github/sanitize.go`: a **surgical byte-pair case** drops `0xC2`
  followed by `0x80`–`0x9F` (the exact UTF-8 encoding of U+0080–U+009F),
  consuming both bytes. Deliberately *not* a rune-based rewrite, to preserve
  the current byte-for-byte handling of all other input (including invalid
  UTF-8). Docstring corrected to match.
- `internal/github/sanitize_test.go`: `TestSanitizeC1ControlUTF8` now asserts
  C1 is **stripped**, with boundary cases (U+0080 / U+009F), an
  8-bit-CSI-neutered-to-literal-text case, and a `U+00A0 (just past C1)
  preserved` regression guard that proves the fix doesn't over-reach into the
  adjacent printable range (U+00A0 shares the `0xC2` lead byte).

## Severity

Low in practice — modern UTF-8 terminals don't interpret `0xC2 0x9B` as the
8-bit CSI introducer — but a security boundary should keep the contract it
documents. The test's safety check confirms no raw C0/DEL byte ever leaks
through.

## Verification

- `go test -race ./...` — green (including the new C1 cases)
- `go vet ./...`, `gofmt -l .` — clean
- `govulncheck ./...` — clean (the CI gate, now on `main`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced text sanitization to properly strip UTF-8-encoded control characters, improving security and preventing potential display issues from malformed input sequences.

* **Tests**
  * Updated and expanded test coverage to verify correct handling of edge-case control character encodings.

### CODE REVIEW

File: internal/github/sanitize.go [hunk 1]
```diff
@@ -27,8 +27,10 @@ import (
 //     sequences (covers the common attack shapes).
 //  2. The byte-level filter removes any remaining C0 control
 //     characters (0x00–0x1F) plus DEL (0x7F), keeping only
-//     newline and tab. C1 controls inside multi-byte UTF-8
-//     sequences are already covered by the ansi pass.
+//     newline and tab. UTF-8-encoded C1 controls (U+0080–U+009F,
+//     the 8-bit CSI/OSC/DCS introducers) are dropped by the
+//     byte-pair case below — ansi.Strip only handles the 7-bit
+//     ESC-prefixed forms.
```
[the remark this answers is not in the document]
Comment (gfazioli) [reply internal/github/sanitize.go:33]: Good catch, and it matches a follow-up already noted in this change's plan. This PR is deliberately scoped to the **primary** boundary — `github.Sanitize`, the extractor boundary where every GitHub-sourced string is sanitized at fetch time. `internal/ui/markdown.go`'s `sanitizeBody` is the defense-in-depth *second* layer: since all content already passes through `github.Sanitize` (which now strips C1) before it reaches `sanitizeBody`, this isn't a live exposure — but the docstring is now inconsistent and the symmetry should be completed. Tracking it as a dedicated follow-up (docstring fix + the same C1 byte-pair filter + a test, as markdown.go currently has none). Minor note: leaving `0xC2 0x9B` intact is valid UTF-8 (U+009B), not invalid UTF-8.
